source/Octopus.Client/Repositories/Async/MixScopeResourceRepository.cs

```diff
@@ -0,0 +1,36 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Octopus.Client.Extensibility;
+using Octopus.Client.Util;
+
+namespace Octopus.Client.Repositories.Async
+{
+    class MixScopeResourceRepository<TMixScopedResource> : BasicRepository<TMixScopedResource> where TMixScopedResource : class, IResource
+    {
+        public MixScopeResourceRepository(IOctopusAsyncClient client, string collectionLinkName)
+            : base(client, collectionLinkName)
+        {
+        }
+
+        public async Task<List<TMixScopedResource>> Search(bool includeGlobal, string[] spaceIds, object parameters = null)
```

Comment:
There's more than just `Search` which may need to be scoped to certain spaces/includeGlobal. It really applies for any get/delete.

What about an API that's more like this:

`repository.Teams.LimitTo("spaces-1", "spaces-2").FindAll()`

Any mixed scope repository type would have this `LimitTo` method on it.

Reply:
very good idea